README.md

```diff
@@ -1,2 +1,105 @@
 # sampler
 public infrastructure tools
+
+
+## Overview
+
+This repository provides a Bash script 
+(`get-aws-subaccount-session.sh`) to establish 
+an AWS CLI session with a subaccount using 
+role assumption. It allows users to securely 
+switch AWS accounts using MFA authentication, 
+ensuring temporary access without storing 
+permanent credentials.
+
+The script:
+* Assumes an IAM role in a subaccount using 
+AWS STS.
+* Requires MFA authentication to generate 
+temporary session tokens.
+* Supports manual or automated MFA entry 
+(if a TOTP secret is stored).
+* Ensures AWS credentials persist in the 
+terminal session.
+
+### Prerequisites
+
+Before using this script, ensure that:
+* You have AWS CLI installed and configured 
+with your base account credentials.
+* Your IAM user has permissions to assume 
+a role in the subaccount.
+* You have an MFA device registered for 
+authentication.
+
+### Usage
+1. Make the Script Executable (Only Once)
+```bash
+chmod +x aws-cli/get-aws-subaccount-session.sh
+```
```

Comment:
This step is not necessary. When sourcing a file, it is just text passed to the current shell.